tools/mmp/driver.cs

```diff
@@ -1495,6 +1499,22 @@ static void CreateSymLink (string directory, string real, string link)
 
 				CopyResourceFile ("config", "config");
 			}
+
+			string machineConfigDestDir = Path.Combine (mmp_dir, "mono/4.5/");
+			string machineConfigDestFile = Path.Combine (machineConfigDestDir, "machine.config");
+			if (custom_machine_config_path != null) {
+				if (!File.Exists (custom_machine_config_path))
+					throw new MonoMacException (97, true, "machine.config file '{0}' can not be found.", custom_machine_config_path);
+				CreateDirectoryIfNeeded (machineConfigDestDir);
+				File.Copy (custom_machine_config_path, machineConfigDestFile);
+
+			}
+			else if (!disable_empty_machine_config) {
```

Comment:
Actually, given https://github.com/mono/mono/pull/3715 I think i'll do it this way:

Not set = don't use empty config (since new mono will fix needing it)
Set to empty string = generate an empty config (in case the fix doesn't take for your use case).
Set to valid file = use that file
Set to invalid non String.Empty file = Exception